kv/txn_coord_sender.go

```diff
@@ -758,28 +756,26 @@ func (tc *TxnCoordSender) heartbeat(id string) {
 					Reply: reply,
 				}
 
-				task := tc.stopper.StartTask()
-				if !task.Ok() {
+				if !tc.stopper.RunTask(func() {
```

Comment:
too bad we have a fairly large func closure here that doesn't readily factor out (as something meaningful), but we can probably do something with it.

I agree that that `continue` down there looks weird. Probably worth taking a look at refactoring it away, maybe we never actually have to use the return value in practice (at which point we can drop it).